src/utils/getFlowTypeFromReactComponent.js

```diff
@@ -51,7 +50,48 @@ export default (path: NodePath): ?NodePath  => {
   if (typePath && types.GenericTypeAnnotation.check(typePath.node)) {
     typePath = resolveToValue(typePath.get('id'));
     if (
-      !typePath || 
+      !typePath ||
+      types.Identifier.check(typePath.node) ||
+      isUnreachableFlowType(typePath)
+    ) {
+      return;
+    }
+
+    typePath = typePath.get('right');
+  }
+
+  return typePath;
+}
+
+export function applyToFlowTypeProperties(
+  path: NodePath,
+  callback: (propertyPath: NodePath) => void
+) {
+  if (path.node.properties) {
+    path.get('properties').each(
+      propertyPath => callback(propertyPath)
+    );
+  } else if (path.node.type === 'IntersectionTypeAnnotation') {
+    path.get('types').each(
+      typesPath => applyToFlowTypeProperties(typesPath, callback)
+    );
+  } else if (path.node.type !== 'UnionTypeAnnotation') {
+    // The react-docgen output format does not currently allow
+    // for the expression of union types
```

Comment:
This depends on how exactly we want to go for this. Quoting the flow docs:

> A union type requires for a value to be one of the input types.
> 
> ```js
> /* @flow */
> type U = number | string;
> var x: U = 1;
> x = "two";
> ```
> An intersection type requires a value to be all of the input types:
> 
> ```js
> /* @flow */
> type I = {a: number} & {b: number};
> var x: I = {a: 1, b: 2};
> x = {a: 1, b: 2, c: "three"};
> ```

So intersection was quite straight forward, just merge all objects and the result is the object with all possible props.

With union this is more complex. In [this](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVYAuBPADgUzAFUwBeMAbwEMAuMAOwFcBbAI3wCcBfMAH0tZ0Azpg4BLegHMuAbgwA3KhzAAPOiXLU6ARlkYVZSrTAAiEeKkm9qwxWsHNxk1SEmANGEGmXAEytygA) example, one can see that when using union for prop-types it is one of the objects to be "unioned" (with additional properties allowed).

So in the simple case of 
```
 type U = {a: number} | {a: string, b: string};
```
The resulting docs would need to be: "prop a is required and needs to be a number, or it is a string but then also b is required and needs to be a string"

This might make sense in some react components, but I'm not sure yet how we would document this cases. We maybe could only add the raw value.

P.S.: I might be mistaken here, but I think it should be correct. Never used union and intersect in these cases myself. :)